services/mailservice/mail_test.go

```diff
@@ -300,3 +304,60 @@ func TestAuthMethods(t *testing.T) {
 		})
 	}
 }
+
+type mockMailer struct {
+	data []byte
+}
+
+func (m *mockMailer) Mail(string) error             { return nil }
+func (m *mockMailer) Rcpt(string) error             { return nil }
+func (m *mockMailer) Data() (io.WriteCloser, error) { return m, nil }
+func (m *mockMailer) Write(p []byte) (int, error) {
+	m.data = append(m.data, p...)
+	return len(p), nil
+}
+func (m *mockMailer) Close() error { return nil }
+
+func TestSendMail(t *testing.T) {
+	dir, err := ioutil.TempDir("", "")
```

Comment:
Random comment: using `TempDir` this way will prevent Go from caching the test results, since the working path is outside the package. I only just recently encountered this myself on mattermost-cloud, where the number of tests is so small that a non-cached test stands out.

I fixed it in [this commit](https://github.com/mattermost/mattermost-cloud/pull/18/commits/dd2ceee3172c8bc0b74de38ee695ef1f11290cde) by using the current directory instead: `ioutil.TempDir(".", "k8s-file-testing-")`, with the second parameter helping to name the directory in case something prevents its cleanup.)

The only downside is that if the `os.RemoveAll` never runs for some reason, there's a garbage temp directory sitting around. 0/5 on whether we should do this here, but throwing it out there for discussion :)

Reply:
Interesting. You're right it was preventing cacheing. Thanks!